indico/queries/submission.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Dict, List
+
+from indico.client.request import GraphQLRequest
+from indico.types import Job, Submission
+
+
+class ListSubmissions(GraphQLRequest):
+    query = """
+        query ListSubmissions($submissionIds: [Int], $workflowIds: [Int], $filters: SubmissionFilter, $limit: Int){
+	        submissions(submissionIds: $submissionIds, workflowIds: $workflowIds, filters: $filters, limit: $limit){
+                submissions {
+                    id
+                    datasetId
+                    workflowId
+                    status
+                    inputFile
+                    inputFilename
+                    resultFile
+                }
+            }
+        }
+    """
+
+    def __init__(
+        self,
+        submission_ids: List[int] = None,
+        workflow_ids: List[int] = None,
+        filters: Dict[str, Any] = None,
+        limit: int = None,
+    ):
+        super().__init__(
+            self.query,
+            variables={
+                "submissionIds": submission_ids,
+                "workflowIds": workflow_ids,
+                "filters": json.dumps(filters) if filters else None,
+                "limit": limit,
+            },
+        )
+
+    def process_response(self, response) -> List[Submission]:
+        return [
+            Submission(**s)
+            for s in super().process_response(response)["submissions"]["submissions"]
+        ]
+
+
+class SubmissionResult(GraphQLRequest):
+    query = """
+        mutation CreateSubmissionResults($submissionId: Int!) {
+            submissionResults(submissionId: $submissionId) {
+                jobId
+            }
+        }
+
+    """
+
+    def __init__(
+        self, submission_id: int,
+    ):
+        super().__init__(
+            self.query, variables={"submissionId": submission_id},
+        )
+
+    def process_response(self, response) -> Job:
+        response = super().process_response(response)["submissionResults"]
+        return Job(id=response["jobId"])
```

Comment:
do we need a request chain here?
like - createSubmissionResults followed by polling for the returned async job?

Reply:
:man_shrugging: @astha seems like the polling for the job is part of the core client functionality. We could do that for users, but that seems pretty easy to do programatically as a user -- Thought I wouldn't say I have any context at all about what level of abstraction we're shooting for with the python client

Reply:
oh cool! nah, you know more about the SDK than I do. this is solid then :+1: 
do you think we need some `SubmissonStatus` child class of JobStatus so we can poll for submission ids after a submission is created?

Reply:
Normally we add a wait flag to allow users to make it blocking.